ThunkLibs/libX11/libX11_Host.cpp

```diff
@@ -25,103 +25,301 @@ desc: Handles callbacks and varargs
 #include <utility>
 
 #include "ldr_ptrs.inl"
+#ifdef _M_ARM_64
+__attribute__((naked))
+void* InternalVariadic(void* Ptr, uint64_t a_0, size_t count, void** list) {
+  // Arguments 0-7 is in registers
+  // 8+ stored on to stack
+  // 0 need to be on the end of the list
+  //
+  // Incoming:
+  // x0 = Function
+  // x1 = XIM
+  // x2 = count
+  // x3 = stack array of 64-bit values
+  //
+  // Outgoing:
+  // x0: Ptr
+  __asm volatile(R"(
+    # Store LR and frame pointer if it exists
```

Comment:
Does this produce significantly better results than a switch case where only the `count > 6` case is implemented as inline-assembly?

Reply:
Interestingly, depending on the branch, the assembly is more or less efficient depending.

C code is generally more efficient for the 1-6 case but only because it does a tail call optimization, adding that to the assembly is easy.
The variadic call ends up poisoning the rest of the function requiring the stack to be adjusted for the entire thing.

I'll shuffle things around so this gets inlined for the full asm to remove the jump, which will make the inline asm more optimal overall.

Reply:
Ah, bummer this doesn't work out in practice.